restcomm/restcomm.dao/src/main/java/org/mobicents/servlet/restcomm/dao/mybatis/MybatisGeolocationDao.java

```diff
@@ -0,0 +1,213 @@
+ *
+ */
+public class MybatisGeolocationDao implements GeolocationDao {
+
+    private static final String namespace = "org.mobicents.servlet.sip.restcomm.dao.GeolocationDao.";
+    private final SqlSessionFactory sessions;
+
+    public MybatisGeolocationDao(final SqlSessionFactory sessions) {
+        super();
+        this.sessions = sessions;
+    }
+
+    @Override
+    public void addGeolocation(Geolocation gl) {
+        final SqlSession session = sessions.openSession();
+        try {
+            session.insert(namespace + "addGeolocation", toMap(gl));
+            session.commit();
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public Geolocation getGeolocation(Sid sid) {
+        return getGeolocation(namespace + "getGeolocation", sid.toString());
+    }
+
+    private Geolocation getGeolocation(final String selector, final String parameter) {
+        final SqlSession session = sessions.openSession();
+        try {
+            final Map<String, Object> result = session.selectOne(selector, parameter);
+            if (result != null) {
+                return toGeolocation(result);
+            } else {
+                return null;
+            }
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public List<Geolocation> getGeolocations(Sid accountSid) {
+        final SqlSession session = sessions.openSession();
+        try {
+            final List<Map<String, Object>> results = session.selectList(namespace + "getGeolocations", accountSid.toString());
+            final List<Geolocation> geolocations = new ArrayList<Geolocation>();
+            if (results != null && !results.isEmpty()) {
+                for (final Map<String, Object> result : results) {
+                    geolocations.add(toGeolocation(result));
+                }
+            }
+            return geolocations;
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public void removeGeolocation(Sid sid) {
+        removeGeolocations(namespace + "removeGeolocation", sid);
+    }
+
+    @Override
+    public void removeGeolocations(final Sid accountSid) {
+        removeGeolocations(namespace + "removeGeolocations", accountSid);
+    }
+
+    private void removeGeolocations(final String selector, final Sid sid) {
+        final SqlSession session = sessions.openSession();
+        try {
+            session.delete(selector, sid.toString());
+            session.commit();
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public void updateGeolocation(Geolocation gl) {
+        final SqlSession session = sessions.openSession();
+        try {
+            session.update(namespace + "updateGeolocation", toMap(gl));
+            session.commit();
+        } finally {
+            session.close();
+        }
+    }
+
+    private Map<String, Object> toMap(Geolocation gl) {
```

Comment:
maybe you can create a class to isolate this two method loc<->map

also map keys should be constants and reused